### PR TITLE
loadfile: fix an old wonky playlist heuristic

### LINKED ISSUE #6576: playlist-prev jumps multiple files
something related to `playback_short` results in `playlist-prev` jumping multiple files even when it shouldnt.

### LINKED ISSUE #12548: playlist-prev skips files after seeking over short videos too quickly.
### Important Information

- mpv version - `mpv v0.36.0-455-ge9690727`
- Windows Version - `WIN10 21H2`
- Source of the mpv binary `https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/`
- ~~If known which version of mpv introduced the problem~~
- Possible screenshot or video of visual glitches:

https://github.com/mpv-player/mpv/assets/77424331/7c354b45-56ff-4f09-8e0a-79988241a962

https://github.com/mpv-player/mpv/assets/77424331/2e854012-0dad-4230-b8ca-aab39dae7d55

### Reproduction steps

- Create a playlist containing at least 1 video that's <2min long (sample files included below).
- Hold `up arrow` to seek over the video and keep holding until MPV goes over the short videos.
- Use `playlist-prev` to toggle through, back to the first video.

### Expected behavior
`playlist-prev` should go back through every video in playlist rather than skipping them like they don't exist anymore.

### Actual behavior
The moment you skip over a short video, `playlist-prev` stops detecting videos and loads files out of order, while `playlist-next` still goes properly through all of them, additionally i had scenarios where the player closes out of nowhere while seeking over, but i'm uncertain if this is connected.

### Log file
[mpvLogFile.txt](https://github.com/mpv-player/mpv/files/12788719/mpvLogFile.txt)

### Sample files
[playlist-prev-test-samples.zip](https://github.com/mpv-player/mpv/files/12788674/playlist-prev-test-samples.zip).

### PR DESCRIPTION
**Pull request by Dudemanguy**
2c6a3cb1f27e02b2e66390a2465ab648905a64d0 originally added this struct member and then 1be863afdbe9017aa227234e8525b209fb818224 later added some more logic to loadfile that uses this. There's been more changes since then of course, but bits using playback_short and playback_start have mostly stayed the same. It's a bit strange it's worked this way for so long since it makes an assumption on how long files should be and leads to weird, broken behavior on playlists with shorter videos. The main reason for playlist_short, as far as I can tell, is to deal with some fringe cases with short videos and trying to go back in the playlist. More specifically, if you use --loop=inf on a very short video (say less than 1 second) and try to go back in the playlist, you won't be able to without any of this special logic that deals with it. But the current approach has several side effects like going back multiple items in the playlist instead of just one if the video is less than one second. This is just bad so delete everything related to playlist_short and playlist_start.

Instead, let's handle this by keeping track of playlist-prev attempts. Going forward in the playlist doesn't require any special handling since a bad/broken file will just advance to the next one. So it's only going backwards that requires some special consideration. If we're going backwards and the user isn't using force, then mark the playlist entry with a special flag. If the file loads successfully in play_current_file, we can just clear the flag and not worry about it. However if there's a failure, then we set a bool telling play_current_file that it should go back one more item in the playlist if possible and try again. This way, we avoid the previously mentioned --loop=inf edgecase and the user can still attempt to retry previously failed items in the playlist (like a url or such).

Fixes #6576, fixes #12548.

### CODE REVIEW

File: common/playlist.h [hunk 1]
```diff
@@ -43,11 +43,9 @@ struct playlist_entry {
     // Used for unshuffling: the pl_index before it was shuffled. -1 => unknown.
     int original_index;
 
-    // Set to true if playback didn't seem to work, or if the file could be
-    // played only for a very short time. This is used to make playlist
-    // navigation just work in case the user has unplayable files in the
-    // playlist.
-    bool playback_short : 1;
+    // Set to true if we're trying to go backwards in the playlist.
+    bool playlist_prev_attempt;
```
Comment (sfan5) [inline common/playlist.h:47]: missing `: 1` to keep the bitfield "optimization"

**Comment by Dudemanguy** [comment]
We discovered in IRC that `mp_next_file` actually has an extra argument that is not actually used so I added another commit that removes it.

File: common/playlist.h [hunk 2]
```diff
@@ -43,11 +43,9 @@ struct playlist_entry {
     // Used for unshuffling: the pl_index before it was shuffled. -1 => unknown.
     int original_index;
 
-    // Set to true if playback didn't seem to work, or if the file could be
-    // played only for a very short time. This is used to make playlist
-    // navigation just work in case the user has unplayable files in the
-    // playlist.
-    bool playback_short : 1;
+    // Set to true if we're trying to go backwards in the playlist.
```
Comment (sfan5) [inline common/playlist.h:46]: on the current entry? on the to-be-tried entry?
what happens if it's set?
(the comment should ideally answer this)

Comment (Dudemanguy) [reply common/playlist.h]: It was a bit terse. Hopefully better now.

**Review by sfan5 — APPROVED** [review]

**Pull request merged**